lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,458 @@
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type GenerateAzureTfConfigurationArgs struct {
+	// Should we configure Activity Log integration in LW?
+	ActivityLog bool
+
+	// Should we add Config integration in LW?
+	Config bool
+
+	// Should we create an Active Directory integration
+	CreateAdIntegration bool
+
+	// If Config is true, give the user the opportunity to name their integration. Defaults to "TF Config"
+	ConfigIntegrationName string
+
+	// If ActivityLog is true, give the user the opportunity to name their integration. Defaults to "TF activity log"
+	ActivityLogIntegrationName string
+
+	// Active Directory application Id
+	AdApplicationId string
+
+	// Active Directory password
+	AdApplicationPassword string
+
+	// Active Directory Enterprise app object id
+	AdServicePrincipalId string
+
+	// Should we use the management group, rather than subscription
+	ManagementGroup bool
+
+	// Management Group ID to set
+	ManagementGroupId string
+
+	// List of subscription Ids
+	SubscriptionIds []string
+
+	// Grant read access to ALL subscriptions
+	AllSubscriptions bool
+
+	// Storage Account name
+	StorageAccountName string
+
+	// Storage Account Resource Group
+	StorageAccountResourceGroup string
+
+	// Should we use existing storage account
+	ExistingStorageAccount bool
+
+	// Azure region where the storage account for logging resides
+	Location string
+}
+
+// Ensure all combinations of inputs are valid for supported spec
+func (args *GenerateAzureTfConfigurationArgs) validate() error {
+	// Validate one of config or activity log was enabled; otherwise error out
+	if !args.ActivityLog && !args.Config {
+		return errors.New("audit log or config integration must be enabled")
+	}
+
+	// Validate that active directory settings are correct
+	if !args.CreateAdIntegration && (args.AdApplicationId == "" || args.AdServicePrincipalId == "" || args.AdApplicationPassword == "") {
+		return errors.New("Active directory details must be set")
+	}
+
+	// Validate the Mangement Group
+	if args.ManagementGroup && args.ManagementGroupId == "" {
+		return errors.New("When Group Management is enabled, then Group Id must be configured")
+	}
+
+	// Validate Storage Account
+	if args.ExistingStorageAccount {
+		if args.StorageAccountName == "" || args.StorageAccountResourceGroup == "" {
+			return errors.New("When using existing storage account, storage account details must be configured")
+		}
+	}
+
+	return nil
+}
+
+type AzureTerraformModifier func(c *GenerateAzureTfConfigurationArgs)
+
+// NewTerraform returns an instance of the GenerateAzureTfConfigurationArgs struct with the provided enabled
+// settings (config/activity log).
+//
+// Note: Additional configuration details may be set using modifiers of the AzureTerraformModifier type
+//
+func NewTerraform(enableConfig bool, enableActivityLog bool, createAdIntegration bool, mods ...AzureTerraformModifier) *GenerateAzureTfConfigurationArgs {
```

Comment:
Similar to a comment from @ipcrm. I'm not convinced that we need to have `createAdIntegration bool` as part of the constructor.
If we changed the var to `useExistingAdApplication` we can infer whether we need to include the AzureAd module in the TF

Reply:
His comment was that this should be a constructor, I have changed it be that rather than a modifier, do you think it should go back to what it was?